server/src/main/java/org/opensearch/index/translog/TranslogReader.java

```diff
@@ -118,7 +129,18 @@ public Long getCheckpointChecksum() {
     public static TranslogReader open(final FileChannel channel, final Path path, final Checkpoint checkpoint, final String translogUUID)
         throws IOException {
         final TranslogHeader header = TranslogHeader.read(translogUUID, path, channel);
-        return new TranslogReader(checkpoint, channel, path, header, null);
+
+        // When we open a reader to Translog from a path, we want to fetch the checksum
+        // as that would be needed later on while creating the metadata map for
+        // generation to checksum.
+        Long translogChecksum = null;
+        boolean hasFooter = false;
+        try {
+            translogChecksum = TranslogFooter.readChecksum(path);
```

Comment:
This is a bit confusing. As `readChecksum` returns `null` value of checksum for an older version file that doesn't have footer. Here we are storing this `translogChecksum` as `null` and setting hasFooter to `true` even though the file doesn't have footer.

Reply:
Thanks for the suggestion. I have refactored this block to make it better streamlined. Now we will directly check-
```
        boolean hasFooter = translogChecksum != null;
```